api.py

```diff
@@ -1,49 +1,96 @@
+        user = db_operations.get_user(args["name"])
+        return user
 
 
-def abort_if_user_doesnt_exist(email):
-    abort(404, message="Can't find the user")
+class SignUp(Resource):
 
+    def post(self):
+        args = request.args
+        schema = schemas.SignUpInfoSchema.load(args)
+        state = db_operations.signup(args["email"], args["name"],
+                                     args["password"], args["city"],
+                                     args["street"], args["phoneNumber"])
+        if (state == states.SignupState.NAME_ALREADY_EXIST) \
+                or (state == states.SignupState.EMAIL_ALREADY_EXIST):
+            message = {'message': 'User already exist'}
+            return message, 409
 
-def abort_if_user_exists(email):
-    abort(409, message="User already exist")
+        return 200
 
 
-class User(Resource):
-    def get(self):
-        args = userEmailParser.parse_args()
-        email = args["email"]
-        abort_if_user_doesnt_exist(args["email"])
-        return {"email": email}
+class Login(Resource):
+    def post(self):
+        args = request.args
+        schemas.LoginSchema().load(args)
+        state = db_operations.login(args["email"], args["password"])
+        if state == states.LoginState.INCORRECT_PASSWORD:
+            message = {'message': 'Incorrect password'}
+            return message, 409
+        if state == states.LoginState.USER_NOT_FOUND:
+            message = {'message': 'User does not exist'}
+            return 404
+        return 200
 
+
+class Donee(Resource):
     def post(self):
-        args = userInfoParser.parse_args()
-        abort_if_user_exists(args["email"])
+        args = request.args
+        schemas.DoneeSchema().load(args)
+        state = db_operations.insert_donee(args["id"], args["fname"], args["lname"],
+                                           args["city"], args["street"], args["phone_number"])
+        if state == states.DoneeInsertionState.DONEE_EXISTS:
+            message = {'message': 'Donee already exist'}
+            return message, 409
+        return 200
+
+    @marshal_with(resource_fields.donee_resource_fields)
+    def get(self):
+        args = request.args
+        schemas.DoneeIDSchema.load(args)
+        donee = db_operations.get_donee(args["id"])
+        if donee is None:
+            message = {'message': 'Invalid ID'}
+            return 404
         return args
 
 
-class Login(Resource):
+class Donation(Resource):
     def post(self):
-        args = loginParser.parse_args()
-        email = args["email"]
-        return args
+        args = request.args
+        schemas.DonationSchema.load(args)
+        state = db_operations.insert_donation(args["id"], args["name"],
+                                              args["date"], args["type"],
+                                              args["value"])
+        if state == states.DonationInsertionState.USER_DOESNT_EXIST:
+            message = {'message': 'Invalid User'}
```

Comment:
Just for the sake of consistency, either `"` or `'` should be used for strings literals across the code base.